CodeEdit/Features/Settings/Pages/ThemeSettings/Models/ThemeModel+CRUD.swift

```diff
@@ -0,0 +1,292 @@
+                        $0.name == prefs.theme.selectedDarkTheme
+                    } ?? self.darkThemes.first
+
+                    self.selectedLightTheme = self.lightThemes.first {
+                        $0.name == prefs.theme.selectedLightTheme
+                    } ?? self.lightThemes.first
+
+                    // For selecting the default theme, doing it correctly on startup requires some more logic
+                    let userSelectedTheme = self.themes.first { $0.name == prefs.theme.selectedTheme }
+                    let systemAppearance = NSAppearance.currentDrawing().name
+
+                    if userSelectedTheme != nil {
+                        self.selectedTheme = userSelectedTheme
+                    } else {
+                        if systemAppearance == .darkAqua {
+                            self.selectedTheme = self.selectedDarkTheme
+                        } else {
+                            self.selectedTheme = self.selectedLightTheme
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    func importTheme() {
+        let openPanel = NSOpenPanel()
+        let allowedTypes = [UTType(filenameExtension: "cetheme")!]
+
+        openPanel.prompt = "Import"
+        openPanel.allowedContentTypes = allowedTypes
+        openPanel.canChooseFiles = true
+        openPanel.canChooseDirectories = false
+        openPanel.allowsMultipleSelection = false
+
+        openPanel.begin { result in
+            if result.rawValue == NSApplication.ModalResponse.OK.rawValue {
+                if let url = openPanel.urls.first {
+                    self.duplicate(url)
+                }
+            }
+        }
+    }
+
+    func duplicate(_ url: URL) {
+        do {
+            self.isAdding = true
+            // Construct the destination file URL
+            var destinationFileURL = self.themesURL.appendingPathComponent(url.lastPathComponent)
+
+            // Extract the base filename and extension
+            let fileExtension = destinationFileURL.pathExtension
+
+            var fileName = destinationFileURL.deletingPathExtension().lastPathComponent
+            var newFileName = fileName
+
+            // Check if the file already exists
+            var iterator = 1
+
+            let isBundled = url.absoluteString.hasPrefix(bundledThemesURL?.absoluteString ?? "")
+            let isImporting =
+                !url.absoluteString.hasPrefix(bundledThemesURL?.absoluteString ?? "")
+                && !url.absoluteString.hasPrefix(themesURL.absoluteString)
+
+            if isBundled {
+                newFileName = "\(fileName) \(iterator)"
+                destinationFileURL = self.themesURL
+                    .appendingPathComponent(newFileName)
+                    .appendingPathExtension(fileExtension)
+            }
+
+            while FileManager.default.fileExists(atPath: destinationFileURL.path) {
```

Comment:
Can you double check if we have code like this already somewhere, where we can generate a unique filename by incrementing an iterator? Maybe this should be a function.

Reply:
I didn't see anything.